src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PullRequestHookProcessor.java

```diff
@@ -23,33 +23,174 @@
+                            return Collections.emptyMap();
+                        }
+                        BitbucketRepositoryType type =
+                                BitbucketRepositoryType.fromString(getPayload().getRepository().getScm());
+                        if (type == null) {
+                            LOGGER.log(Level.INFO, "Received event for unknown repository type: {0}",
+                                    getPayload().getRepository().getScm());
+                            return Collections.emptyMap();
+                        }
+                        Map<SCMHead, SCMRevision> result = new HashMap<>(1);
+                        PullRequestSCMHead head = new PullRequestSCMHead(
+                                getPayload().getPullRequest().getSource().getRepository().getOwnerName(),
+                                getPayload().getPullRequest().getSource().getRepository().getRepositoryName(),
+                                type,
+                                getPayload().getPullRequest().getSource().getBranch().getName(),
+                                getPayload().getPullRequest()
+                        );
+                        if (hookEvent == PULL_REQUEST_DECLINED || hookEvent == PULL_REQUEST_MERGED) {
+                            // special case for repo being deleted
+                            result.put(head, null);
+                        } else {
+                            switch (type) {
+                                case GIT:
+                                    result.put(head, new AbstractGitSCMSource.SCMRevisionImpl(head,
+                                            getPayload().getPullRequest().getSource().getCommit().getHash()));
+                                    break;
+                                case MERCURIAL:
+                                    result.put(head, new BitbucketSCMSource.MercurialRevision(head,
+                                            getPayload().getPullRequest().getSource().getCommit().getHash()));
+                                    break;
+                                default:
+                                    LOGGER.log(Level.INFO, "Received event for unknown repository type: {0}", type);
+                                    break;
+                            }
+                        }
+                        return result;
+                    }
 
-                LOGGER.info(String.format("Received hook from Bitbucket. Processing pull request to %s/%s from %s/%s", 
-                        owner, repository, pull.getPullRequest().getSource().getRepository().getOwnerName(), pull.getPullRequest().getSource().getRepository().getRepositoryName()));
-                scmSourceReIndex(owner, repository);
+                    @Override
+                    public boolean isMatch(@NonNull SCM scm) {
+                        // TODO
```

Comment:
We are not implementing this at this point in time because we do not want to introduce the triggering of freestyle jobs based on the bitbucket hook data provided by the branch source registered hooks.

At a future point in time we will add the matching logic in here, but right now it would increase the scope of the change too much